File: data-importer/src/types/review.rs
use chrono::{DateTime, Utc};
use serde::Deserialize;
use uuid::Uuid;

#[derive(Deserialize, Debug, Clone)]
pub enum ReviewStatus {
    #[serde(rename = "approved")]
    Approved,
    #[serde(rename = "pending_approval")]
    PendingApproval,
    #[serde(rename = "removed")]
    Removed,
}

#[derive(Deserialize, Debug, Clone)]
pub struct ReviewScoreAspects {
    pub room: Option<i32>,
    pub food: Option<i32>,
    pub service: Option<i32>,
    pub location: Option<i32>,
    pub hygiene: Option<i32>,
    pub pool: Option<i32>,
    #[serde(rename = "childFriendly")]
    pub child_friendly: Option<i32>,
    #[serde(rename = "priceQuality")]
    pub price_quality: Option<i32>,
}

#[derive(Debug, Deserialize, Clone)]
pub struct Review {
    pub id: Uuid,
    #[serde(rename = "zooverReviewId")]
    pub zoover_review_id: i32,

    #[serde(rename = "accommodationId")]
    pub accommodation_id: Uuid,

    #[serde(rename = "userName")]
    pub user_name: String,

    pub title: Option<String>,
    pub text: String,
    pub status: ReviewStatus,

    #[serde(rename = "generalScore")]
    pub general_score: f32,
    #[serde(rename = "scoreAspects")]
    pub score_aspects: String,

    #[serde(rename = "createdAt")]
    pub created_at: DateTime<Utc>,
    #[serde(rename = "updatedAt")]
    pub updated_at: DateTime<Utc>,
}

File: data-service/app/types/review.py
from datetime import datetime
from enum import Enum
from uuid import UUID

class ReviewStatus(str, Enum):
    APPROVED = "approved"
    PENDING_APPROVAL = "pending_approval"
    REMOVED = "removed"

class ReviewScoreAspects:
    room: int | None
    food: int | None
    service: int | None
    location: int | None
    hygiene: int | None
    pool: int | None
    child_friendly: int | None
    price_quality: int | None

class Review:
    id: UUID
    zoover_review_id: int

    accomodation_id: UUID

    user_name: str

    title: str | None
    text: str
    status: ReviewStatus

    general_score: float
    score_aspects: ReviewScoreAspects
    
    created_at: datetime
    updated_at: datetime

